mne_neuron/dipole.py

```diff
@@ -101,18 +156,56 @@ def prsimtime():
                      np.array(dp_rec_L2.to_python()),
                      np.array(dp_rec_L5.to_python())]
 
-    pc.gid_clear()
-    pc.done()
 
     dpl = Dipole(np.array(t_vec.to_python()), dpl_data)
 
+    err = None
     if rank == 0:
+        if net.params['save_dpl']:
+            dpl.write('rawdpl_%d.txt' % trial)
+
         dpl.baseline_renormalize(net.params)
         dpl.convert_fAm_to_nAm()
         dpl.scale(net.params['dipole_scalefctr'])
         dpl.smooth(net.params['dipole_smooth_win'] / h.dt)
 
-    return dpl
+        try:
+            if extdata.any():
+                ddat = {'dpl' : dpl.dpl, 'dextdata' : extdata}
+                err = calcerr(ddat)
```

Comment:
it seems a bit like you want to kwargs-style dictionary but I would rather suggest keeping it simple and passing two `Dipole` objects.